backend/api/Services/InspectionAreaService.cs

```diff
@@ -127,6 +134,83 @@ public async Task<IEnumerable<InspectionArea>> ReadByInstallation(
                 .FirstOrDefaultAsync();
         }
 
+        public bool MissionTasksAreInsideInspectionAreaPolygon(
+            List<MissionTask> missionTasks,
+            InspectionArea inspectionArea
+        )
+        {
+            if (string.IsNullOrEmpty(inspectionArea.AreaPolygonJson))
+            {
+                return true; // No polygon defined, so all tasks are inside
+            }
+
+            var inspectionAreaPolygon = JsonSerializer.Deserialize<InspectionAreaPolygon>(
+                inspectionArea.AreaPolygonJson
+            );
+
+            if (inspectionAreaPolygon == null)
+            {
+                return true; // Invalid polygon, so all tasks are inside
```

Comment:
Should this trigger a warning before returning?

Reply:
Would probably be best yeah =)